emily/handler/src/api/handlers/chainstate.rs

```diff
@@ -169,6 +174,27 @@ pub async fn add_chainstate_entry_or_reorg(
     context: &EmilyContext,
     chainstate: &Chainstate,
 ) -> Result<(), Error> {
+    // We don't want to do reorg for too old chainstates, because it highly
+    // likely means that we reconnected to new stacks node which is not fully synced,
+    // rather then actual reorg happen.
+    let new_bitcoin_tip_height = chainstate.bitcoin_block_height;
+    let current_bitcoin_tip_height = accessors::get_api_state(context)
+        .await?
+        .chaintip()
+        .bitcoin_height;
+
+    if let Some(current_bitcoin_tip_height) = current_bitcoin_tip_height {
```

Comment:
You should be able to reduce nesting with `if let (Some(current_height), Some(new_height)) = (current_bitcoin_tip_height, new_bitcoin_tip_height) { ... }`

Reply:
@cylewitruk